cnxepub/html_parsers.py

```diff
@@ -16,6 +16,15 @@
     }
 
 
+def _squash_to_text(elm):
+    value = [elm.text or '']
+    for child in elm.getchildren():
+        value.append(etree.tostring(child).decode('utf-8'))
+        value.append(child.tail or '')
+    value = ''.join(value).encode('utf-8')
+    return value
+
+
```

Comment:
One of the tests failed last time we tried. Maybe there was a different
problem... Do you want to try running the tests with what you think will
work?
On May 19, 2016 14:42, "Ross Reedstrom" <notifications@github.com> wrote:

> In cnxepub/html_parsers.py
> <https://github.com/Connexions/cnx-epub/pull/73#discussion_r63956018>:
>
> > @@ -16,6 +16,15 @@
> >      }
> >
> >
> > +def _squash_to_text(elm):
> > +    value = [elm.text or '']
> > +    for child in elm.getchildren():
> > +        value.append(etree.tostring(child).decode('utf-8'))
> > +        value.append(child.tail or '')
> > +    value = ''.join(value).encode('utf-8')
> > +    return value
> > +
> > +
>
> >>> e.text=u'This is élément'
> >>> etree.tostring(e1)
> '<div><a>This is &#233;l&#233;ment</a></div>'
> >>> etree.tostring(e1, method='html', encoding='utf-8')
> '<div><a>This is \xc3\xa9l\xc3\xa9ment</a></div>'
>
> One of those should work, no?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Connexions/cnx-epub/pull/73/files/053a264da5e2e901d4f35682e7c2a52831f4f6d0#r63956018>
>